src/vehicle.cpp

```diff
@@ -6337,20 +6337,25 @@ bool vehicle::fire_turret( int p, bool manual )
         charges = std::min(charges, turret_data.charges);
     }
 
-    // Create a fake gun
-    // @todo damage the gun based on part hp
-    item gun( turret_data.gun.typeId(), turret_data.gun.bday );
+    auto gun = item_location( vehicle_cursor( *this, p ), &parts[ p ].base );
     if( turret_data.gun.ammo_current() != "null" ) {
-        gun.ammo_set( turret_data.gun.ammo_current(), charges );
+        gun->ammo_set( turret_data.gun.ammo_current(), charges );
     }
 
     // TODO sometime: change that g->u to a parameter, so that NPCs can shoot too
     // TODO: unify those two functions.
-    int shots = manual ? manual_fire_turret( p, g->u, gun ) : automatic_fire_turret( p, gun );
-    turret_data.consume( *this, p, shots * gun.ammo_required() );
+    int qty = gun->ammo_required();
+    int ups = gun->get_gun_ups_drain();
+
+    gun->item_tags.insert( "TURRET" );
+    int shots = manual ? manual_fire_turret( p, g->u, std::move( gun ) ) : automatic_fire_turret( p, std::move( gun ) );
+    gun->item_tags.erase( "TURRET" );
```

Comment:
Makes sense, I missed bionic guns. Then why the flag can't be added permanently?